main/src/library/pkger/Manifest.flix

```diff
@@ -0,0 +1,287 @@
+namespace Manifest {
+    pub enum License {
+        case MIT
+        case GPLv3
+        case Apache2
+        case BSD
+        case Other(String)
+    }
+
+    pub type alias Author = { name: String
+                            // Is there an email datatype, or is it just commonly a String?
+                            , email: String
+                            }
+
+    pub type alias Package = { name: String
+                             , version: SemVer.SemVer
+                             , authors: List[Author]
+                             , description: String
+                             // This could be mandatory
+                             , license: Option[License]
+                             // Mapping package dependency names to the semver pattern
+                             , depends: Map[String, SemVer.SemVer]
+                             , home: Option[String] // Url?
+                             }
+
+    pub type alias LockedPackage = { name: String
+                                   , version: SemVer.SemVer
+                                   // All transitive dependencies, their specific version, and where they were retrieved from.
+                                   , transitive: Map[String, SemVer.SemVer]
+                                   }
+
+    def newLockedPackage(name: String, version: SemVer.SemVer, transitive: Map[String, SemVer.SemVer]): LockedPackage = {
+        {name = name, version = version, transitive = transitive}
+    }
+
+    // Add a new package to the package. Handle superseding old versions
```

Comment:
In Flix, comments are of the form:

///
/// Some doc
///

like rust :)

Reply:
I wrote that comment before I started implementing it, as a part of sketching out the structure. But you're right, it should be updated to be a proper doc comments!